packages/zcli-apps/src/commands/apps/new.ts

```diff
@@ -51,12 +53,23 @@ export default class New extends Command {
     })
   }
 
+  async extractSubfolderToTarget (scaffoldName: string, targetName: string) {
+    return new Promise((resolve, reject) => {
+      fsExtra.copy(
+        path.join(process.cwd(), '/', scaffoldName, '/packages/', targetName),
+        path.join(process.cwd(), '/', `${scaffoldName}-${targetName}`),
+        { overwrite: true, errorOnExist: true }, (err) => {
+          if (err) {
+            reject(err)
+          }
+          resolve()
+        }
+      )
+    })
+  }
+
   getScaffoldDir (flagScaffold: string): string {
-    const scaffolds: Scaffolds = {
-      basic: 'apps_scaffold_basic',
-      react: 'app_scaffold'
-    }
-    const scaffoldRepo = scaffolds[flagScaffold]
+    const scaffoldRepo = 'app_scaffolds'
```

Comment:
this change here makes this function useless, previously we were validating if user selection matches the available scaffold but given we removed them from here, it will now always pass even when we pass a wrong scaffold flag.

I think the design of this commands needs tweaking according to this change. I suggest we not only fix this validation but also make this dynamic to avoid updating zcli everytime we add a scaffold.

We do this flow in this command now,

- We download scaffolds repo
```
await this.downloadScaffoldsRepo(`HARD CODED URL`) // we don't need dynamic url anymore, also func is renamed
```
- We check if this scaffold exists that user is selecting, if not we throw a error
```
await scaffoldExists(pathToDownloadedRepo, scaffoldName) /// we validate and throw error if wrong scaffold selected
```

rest of the flow remains the same

Reply:
```
    const scaffoldRepo = this.getScaffoldDir(flagScaffold)
    const scaffoldDir = scaffoldRepo + '-master'
    const scaffoldDirWithType = scaffoldDir + '-' + flagScaffold
```

these have become slightly confusing, with above change we get rid of these too.

Reply:
I've made an attempt at addressing your comments with this commit https://github.com/zendesk/zcli/pull/65/commits/e96fe980fdba7ca4d54c6ea5ad97182663fc704c 🙏

Reply:
Ah, maybe I need to throw an error per comment the first, let me look into that.